framework/src/main/java/org/checkerframework/framework/type/BoundsInitializer.java

```diff
@@ -92,11 +93,16 @@ public static void initializeTypeArgs(AnnotatedDeclaredType declaredType) {
             typeArgs.add(typeArg);
 
             // Add mapping from type parameter to the annotated type argument.
-            typeArgMap.put((TypeVariable) typeElement.getTypeParameters().get(i).asType(), typeArg);
+            typeArgMap.put(
+                    (TypeVariable)
```

Comment:
Please put this long sub-expression into a local variable and add a comment why you're using the un-annotated type.
One pair of parentheses seem superfluous and don't improve readability - remove the pair around the `unannotatedType` method invocation.